pkgs/development/misc/resholve/default.nix

```diff
@@ -0,0 +1,13 @@
+{ callPackage
+, doCheck ? true
+}:
+let
+  resholve = callPackage ./resholve.nix { inherit doCheck; };
+  resholvePackage =
+    callPackage ./resholve-package.nix { inherit resholve; };
```

Comment:
Please move this into ``pkgs/build-support/resholve.nix``.

Reply:
Not keen to add extra friction to developing this out-of-tree in the resholve repo and porting fixes. 

I did go ahead and flatten the namespace out in https://github.com/NixOS/nixpkgs/pull/85827/commits/e9868c1ac77b9420b822a3df354023ffc3c47b7b, which I hope achieves most of your goal here.